src/CMakeLists.txt

```diff
@@ -16,22 +16,73 @@ set(cucumber-src
 )
 
 add_library(cucumber ${cucumber-src} ${CMAKE_CURRENT_SOURCE_DIR}/main.cpp)
-add_library(cucumber-no-main ${cucumber-src})
-
-install(
-	TARGETS cucumber
-    DESTINATION ${CMAKE_INSTALL_PREFIX}
-)
-install(
-	TARGETS cucumber-no-main
-    DESTINATION ${CMAKE_INSTALL_PREFIX}
-)
+add_library(cucumber-no-main ${cucumber-src} )
 
 if (BUILD_SHARED_LIBS)
     set_target_properties(cucumber PROPERTIES
                           POSITION_INDEPENDENT_CODE ON
-                          )
+    )
     set_target_properties(cucumber-no-main PROPERTIES
                           POSITION_INDEPENDENT_CODE ON
-                          )
+    )
 endif()
+
+
+# install(
+#   TARGETS cucumber
+#   COMPONENT cucumber
+# )
+# install(
+#   TARGETS cucumber-no-main
+#   COMPONENT cucumber
+# )
+
+include(CMakePackageConfigHelpers)
+
+set(INCLUDE_INSTALL_DIR "include")
+set(LIBRARY_INSTALL_DIR "lib")
+set(LIBRARY_VERSION ${PROJECT_VERSION})
+set(LIBRARY_SYSTEM_NAME "${CMAKE_HOST_SYSTEM_NAME}")
+
+configure_package_config_file(
+    CwTCucumberConfig.cmake.in ${CMAKE_CURRENT_BINARY_DIR}/CwTCucumberConfig.cmake
+    INSTALL_DESTINATION lib/cwtcucumber/cmake
+    PATH_VARS INCLUDE_INSTALL_DIR LIBRARY_INSTALL_DIR LIBRARY_VERSION LIBRARY_SYSTEM_NAME
+)
+
+write_basic_package_version_file(
+    ${CMAKE_CURRENT_BINARY_DIR}/CwTCucumberConfigVersion.cmake

```

Comment:
I never used the naving convention CwT, please use here cwt-... so it is consistent with the conan package from the package repo